sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -228,4 +230,179 @@ boolean isDone() {
+
+        volatile long requested;
+        @SuppressWarnings("rawtypes")
+        static final AtomicLongFieldUpdater<BaseSubscription> REQUESTED =
+            AtomicLongFieldUpdater.newUpdater(BaseSubscription.class, "requested");
+
+        BaseSubscription(Subscriber<? super U> subscriber, ContinuationState<C> continuationState,
+            PageRetriever<C, P> pageRetriever, Integer defaultPageSize) {
+            this.subscriber = subscriber;
+            this.continuationState = continuationState;
+            this.pageRetriever = pageRetriever;
+            this.defaultPageSize = defaultPageSize;
+        }
+
+        @Override
+        public void request(long l) {
+            if (Operators.validate(l)) {
+                Operators.addCap(REQUESTED, this, l);
+                drain();
+            }
+        }
+
+        private void drain() {
+            if (WIP.getAndIncrement(this) != 0) {
+                return;
+            }
+
+            // On the first request or once the iterator is consumed request a page.
+            if (needToRequestPage()) {
+                requestPage();
+            }
+
+            int missed = 1;
+            while (true) {
+                if (cancelled) {
+                    return;
+                }
+
+                if (REQUESTED.get(this) > 0) {
+                    boolean emitted = false;
+                    // read d before next to avoid race
+                    boolean d = done;
+                    if (hasNext()) {
+                        subscriber.onNext(getNext());
+                        emitted = true;
+                    }
+
+                    if (d) {
+                        if (error != null) {
+                            subscriber.onError(error);
+                        } else {
+                            subscriber.onComplete();
+                        }
+
+                        // exit without reducing wip so that further drains will be NOOP
+                        return;
+                    }
+
+                    if (emitted) {
+                        // do this after checking d to avoid calling read
+                        // when done
+                        Operators.produced(REQUESTED, this, 1);
+                    }
+                }
+
+                missed = WIP.addAndGet(this, -missed);
+                if (missed == 0) {
+                    return;
+                }
+            }
+        }
+
+        abstract boolean needToRequestPage();
+        abstract boolean hasNext();
+        abstract U getNext();
+
+        private void requestPage() {
+            CountDownLatch countDownLatch = new CountDownLatch(1);
+            pageRetriever.get(continuationState.getLastContinuationToken(), defaultPageSize)
+                .singleOrEmpty()
+                .subscribe(this::setNext, this::setError, countDownLatch::countDown);
+
+            try {
+                countDownLatch.await();
```

Comment:
I think the `latch.await()` is something we have to be cautious about. 

Our "drain + subscribe + latch.await" could gets executes in the context of a non-blocking Thread. As an example, say app is running in a single thread, thread_0:

   the drain() is executing in thread_0.
   subscribe() results in queueing a n/w call and return.
   latch.await() executed in thread_0, blocking it for the result.
   scheduled n/w call cannot be completed as thread_0 is waiting it’s a deadlock.

Reactor marks some threads as non-blocking https://projectreactor.io/docs/core/release/api/reactor/core/scheduler/NonBlocking.html.

Reply:
Made it so only `PagedIterable` instances will use the blocking calling.